emsdk.py

```diff
@@ -1014,15 +1012,40 @@ def xcode_sdk_version():
     return subprocess.checkplatform.mac_ver()[0].split('.')
 
 
+def decide_cmake_target_platform(tool):
+  if hasattr(tool, 'arch') and tool.arch == 'aarch64':
+    return 'ARM64'
+  elif hasattr(tool, 'arch') and tool.arch == 'x86_64':
+    return 'x64'
+  elif hasattr(tool, 'arch') and tool.arch == 'x86':
+    return 'x86'
```

Comment:
How about just one `if hasattr(tool, 'arch')` here.

e.g.

```
if hasattr(tool, 'arch'):
   tools_arch_map = {'aarch64': ARM64' ...}
   return tool_arch_map[tool.arch]
// Fall back to host arch. 
if ARCH == 'aarch64':
    return 'ARM64'
  else:
    return 'x64' if tool.bitness == 64 else 'x86'
```
 
Should we call `decide_cmake_host_platform` for the fallback case?

Reply:
The code style has changed and the values fixed according to the source references added. The host platform and target platform values has different syntax as you can see in the sources.